src/Avalonia.Base/Input/Focus/FocusInputDeviceKind.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Avalonia.Input
+{
+    /// <summary>
+    /// Specifies input device types from which input events are received
+    /// </summary>
+    public enum FocusInputDeviceKind
+    {
+        /// <summary>
+        /// No input
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// Mouse input device
+        /// </summary>
+        Mouse,
+
+        /// <summary>
+        /// Touch input device
+        /// </summary>
+        Touch,
+
+        /// <summary>
+        /// Pen input device
+        /// </summary>
+        Pen,
+
+        /// <summary>
+        /// Keyboard input device
+        /// </summary>
+        Keyboard,
+
+        /// <summary>
+        /// Game controller/remote input device
+        /// </summary>
+        GameController
```

Comment:
Is this ported from UWP? Because otherwise I don't really like "GameController" term, because controller by itself isn't really limited to games. And also same means "remote controller"  for TV-like devices, basically just a variation of console-controllers.

On other hand we don't support controllers yet, so it can be added later. Personally I want to see it innAvalonia in the future. 

P.S. I just had a small look. Will review properly on next week.

Reply:
Yes the enum names are from UWP. 'GameController' term comes from the Xbox device family. Since Avalonia isn't tied to any platform, if you have a better name suggestion, I don't think it would hurt to change here. This is also unused at the moment, I left it in for future support.

Reply:
How about just `Controller` ?